packages/vendure-order-client/test/vendure-order-client.spec.ts

```diff
@@ -135,10 +138,31 @@ describe('Vendure order client', () => {
 
   describe('Cart management', () => {
     it('Adds an item to order', async () => {
-      const order = await client.addItemToOrder('T_2', 1);
-      activeOrderCode = order?.code;
-      expect(order?.lines[0].quantity).toBe(1);
-      expect(order?.lines[0].productVariant.id).toBe('T_2');
+      let wasSetToTrue = false;
+      let wasFinallySetToFalse = false;
+      listenKeys(client.$activeOrder, ['loading'], (currentActiveOrderData) => {
+        if (currentActiveOrderData.loading) {
+          wasSetToTrue = true;
+        } else {
+          if (wasSetToTrue) {
+            wasFinallySetToFalse = true;
+          }
+        }
```

Comment:
These else statements could be simplified into:

```suggestion
        } else if (wasSetToTrue) {
            wasFinallySetToFalse = true;
        }
```

BUT, another option would be to not use `listenKeys` at all:

```ts
      expect(client.$activeOrder.value?.loading).toBe(false);
      const promise = client.removeOrderLine('T_1');
      expect(client.$activeOrder.value?.loading).toBe(true);
      await promise;
      expect(client.$activeOrder.value?.loading).toBe(false);
```
It does require a bit of work to refactor, but I do find the code a bit more followable. I leave it up to you, both approaches are ok, but if you choose the `listenKeys` options we need to simplify the if/else statement like above.

I do like the custom `expect` messages, I was actually looking for something like this the other day, didnt know this worked :open_hands: